pandas/tests/indexes/period/test_constructors.py

```diff
@@ -321,7 +321,9 @@ def test_constructor_mixed(self):
     def test_constructor_simple_new(self):
         idx = period_range("2007-01", name="p", periods=2, freq="M")
 
-        with pytest.raises(AssertionError, match="<class .*PeriodIndex'>"):
+        with pytest.raises(
+            TypeError, match="_simple_new expects PeriodArray, got PeriodIndex"
```

Comment:
nitpick: i prefer `msg = "_simple_new ..." on the previous line so the with `pytest.raises ` can be a 1-liner.  not a big deal